checkbox-ng/plainbox/impl/session/remote_assistant.py

```diff
@@ -732,22 +739,4 @@ def exposed_cache_report(self, exporter_id, options):
         exported_stream = SpooledTemporaryFile(max_size=102400, mode='w+b')
         exporter.dump_from_session_manager(self._sa._manager, exported_stream)
         exported_stream.flush()
-        return exported_stream
-
-def _guess_normal_user():
-    _logger.warning("normal_user not supplied via config(s).")
-    for entry in pwd.getpwall():
-        if entry.pw_name == 'ubuntu':
-            _logger.warning("Using `ubuntu` user")
-            return 'ubuntu'
-    with suppress(KeyError):
-        user = pwd.getpwuid(1000).pw_name
-        _logger.warning("Using `%s` user", user)
-        return user
-    with suppress(KeyError):
-        user = pwd.getpwuid(1001).pw_name
-        _logger.warning("Using `%s` user", user)
-        return user
-    raise RuntimeError(
```

Comment:
The new guess_normal_user() does not raise a RuntimeError and I think just returns none if there's no normal_user in configs and no users with uid 1000/1001. So a user foo with uid 1002 would return None. As before in such case better to abort checkbox

Reply:
I moved the checking to where the logic can handle it and propagate up. Line 294 of remote_assistant.py. This should cover that case (and an icoming metabox scenario confirms it)

Reply:
> A corner case not covered I think with the new guess_normal_user() version, a user foo w/ uid 1002.

Does it mean we have to cover 1002 too?
Is there a range we should iterate over?

Reply:
I took 1002 but it could be 1042 for a user lambda as long as it exists. There's no range, 1000 and 1001 are the most common use cases. In this scenario, guess_normal_user would return None where in the current code it raises a RuntimeError

Reply:
The RuntimeError is still getting raised by the calling code tho, unless I missed something.

Reply:
in `check_user_exists` yes but not `guess_normal_user`